pykafka/producer.py

```diff
@@ -535,6 +539,8 @@ def queue_reader():
                 except Exception:
                     # surface all exceptions to the main thread
                     self.producer._worker_exception = sys.exc_info()
+                    if not self.running and not self.slot_available.is_set():
```

Comment:
@emmett9001 , thanks for your review. I read the relevant code, and also consider  then "not self.running"   condition here is unneccessary.  I will fix it soon.